internal/s3pmoxcommon/s3pmoxcommon.go

```diff
@@ -65,6 +65,16 @@ func ListSnapshots(c minio.Client, datastore string, returnCorrupted bool) ([]Sn
 				corrupted:  false,
 			}
 
+			tags, err := S.ReadTags(c)
+			if err == nil {
+				if val, ok := tags["protected"]; ok && val == "true" {
+					S.Protected = true
+					s3backuplog.DebugPrint("Snapshot %s marked as protected (from tags)", S.S3Prefix())
+				}
+			} else {
+				s3backuplog.DebugPrint("Unable to get tags for %s: %s", S.S3Prefix(), err)
+			}
+
```

Comment:
Not an error, but a heads up.
Nesting `if` conditions can lead to hard-to-read code. It is better to use early exit or separate `if` conditions.
```suggestion
			if err != nil {
				s3backuplog.DebugPrint("Unable to get tags for %s: %s", S.S3Prefix(), err)
			}
			
			if err == nil && tags["protected"] == "true" {
			    S.Protected = true
                s3backuplog.DebugPrint("Snapshot %s marked as protected (from tags)", S.S3Prefix())
			}

```